agent/command/exec.go

```diff
@@ -163,13 +159,25 @@ type modifyEnvOptions struct {
 	expansions             util.Expansions
 	includeExpansionsInEnv []string
 	addExpansionsToEnv     bool
+	addToPath              []string
 }
 
 func defaultAndApplyExpansionsToEnv(env map[string]string, opts modifyEnvOptions) map[string]string {
 	if env == nil {
 		env = map[string]string{}
 	}
 
+	if len(opts.addToPath) > 0 {
+		// Prepend paths to the runtime environment's PATH. More reasonable
+		// behavior here would be to respect the PATH env var if it's explicitly
+		// set for the command, but changing this could break existing
+		// workflows, so we don't do that.
+		path := make([]string, 0, len(opts.addToPath)+1)
+		path = append(path, opts.addToPath...)
+		path = append(path, os.Getenv("PATH"))
+		env["PATH"] = strings.Join(path, string(filepath.ListSeparator))
```

Comment:
Does this cover windows case?

Reply:
Not sure if you mean `PATH` or the `filepath.ListSeparator`, but yes, this is compatible with both Windows and Linux. The `filepath.ListSeparator` is set to the correct native value for Windows (`;`) vs. Linux (`:`). And both platforms accept `PATH` to search for executables (as confirmed by the [docs](https://pkg.go.dev/os/exec?GOOS=windows#LookPath)).